### ISSUE TITLE
Results not as expected (Table 1: Long-term forecasting results) 

### ISSUE THREAD

**Issue by BennyTMT** [KimMeen/Time-LLM#48]
I used the default hyper-parameters from the repository and 8 A100 (80G) GPUs, but the results were below expectations. 

For instance, these four results in ETTh1 and ETTh2.

ETTh1 512_96   MAE 0.415  (compared to 0.392 in Table 1)
ETTh1 512_192 MAE 0.428 (compared to 0.418 in Table 1)
ETTh2 512_96  MAE 0.365  (compared to 0.328 in Table 1)
ETTh2 512_192 MAE 0.389 (compared to 0.375 in Table 1)

Could it be that I missed some experimental details, or there are aspects I have not paid attention to?

**Comment by kwuking** [comment]
hi, our results are based on python 3.8 torch 2.0.1, trained for 50 epochs, and the average of multiple runs. The provided script is quick demos. You can refer to our paper and repository setup to try to reproduce the results.

**Comment by BennyTMT** [comment]
I didn't change the number of epochs; it's set to default. None of the 8 experiments I conducted met the expected outcomes, so it's likely not due to multiple computations. 

I've observed that in most experiments, the validation loss didn't decrease with training. 

Could there be a problem somewhere else? And, could you find the training loss Figure for Etth1 512 96?

**Comment by kwuking** [comment]
> Thank you for your reply and your work. but I didn't change the number of epochs; it's set to default. None of the 8 experiments I conducted met the expected outcomes, so it's likely not due to multiple computations.
> 
> I've observed that in most experiments, the validation loss didn't decrease with training.
> 
> Could there be other issues I haven't noticed? And, could you find the training loss Figure for Etth1 512 96?

Thank you very much for your interest in our work and for your efforts in replicating our experimental results. I understand your concerns about the inconsistency in training results, and let me explain some of the reasons that might cause this situation.

Firstly, our training process utilizes the ZeRO-2 optimizer under the DeepSpeed framework, which is a technology optimized and accelerated for large-scale training. During this process, we use gradient accumulation to simulate a larger batch size, which helps us manage limited hardware resources but also introduces randomness, as gradient accumulation affects weight updates.

Secondly, we have adopted mixed-precision training, specifically using bfloat16 (bf16) precision, which can significantly reduce memory usage and speed up training. However, employing lower precision in floating-point representation also introduces additional numerical calculation errors, which is another reason for the randomness in results.

Please note that these techniques are very common and necessary for training large models in modern deep learning frameworks, even though they might cause slight fluctuations in results. Additionally, the deep learning runtime environment, version of the framework, version of CUDA, and similar factors could introduce a certain level of randomness.

If you have more specific questions or need assistance, please feel free to contact us via email for further discussion.

Thank you again for your interest and support.
